Src/ILGPU.Tests.CPU/RuntimeTests.cs

```diff
@@ -54,6 +59,16 @@ internal static void TestCustomDeviceSetup_ExplicitKernel(ArrayView<int> data)
                 numThreadsPerWarp,
                 numWarpsPerMultiprocessor,
                 numMultiprocessors);
+
+            // Skip specific tests on MacOS
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // Detect the number of processors and check whether we run in a supported
+                // range of tests
+                int maxNumThreads = Environment.ProcessorCount * 768;
```

Comment:
The actual maximum number of threads on macOS can be found by shelling out this command:
```
sysctl -n kern.num_taskthreads
```

Or even better, you can P/Invoke `sysctl` directly. See https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/sysctl.3.html for the function documentation.